src/cdk-experimental/popover-edit/table-directives.ts

```diff
@@ -266,6 +277,60 @@ export class CdkPopoverEdit<C> implements AfterViewInit, OnDestroy {
   }
 }
 
+/**
+ * Attaches an ng-template to a cell and shows it when instructed to by the
+ * EditEventDispatcher service.
+ * Makes the cell focusable.
+ */
+@Directive({
+  selector: '[cdkPopoverEdit] [cdkPopoverEditTabOut]',
+  host: {
+    'tabIndex': '0',
+    'class': 'cdk-popover-edit-cell',
+    '[attr.aria-haspopup]': 'true',
+  }
+})
+export class CdkPopoverEditTabOut<C> extends CdkPopoverEdit<C> {
+  protected focusTrap?: FocusEscapeNotifier;
+
+  constructor(
+      editEventDispatcher: EditEventDispatcher,
+      elementRef: ElementRef,
+      focusTrapFactory: FocusTrapFactory,
+      ngZone: NgZone,
+      overlay: Overlay,
+      positionFactory: PopoverEditPositionStrategyFactory,
+      scrollDispatcher: ScrollDispatcher,
+      viewContainerRef: ViewContainerRef,
+      viewportRuler: ViewportRuler,
+      protected readonly focusEscapeNotifierFactory: FocusEscapeNotifierFactory,
+      protected readonly focusDispatcher: FocusDispatcher) {
+    super(
+        editEventDispatcher,
+        elementRef,
+        focusTrapFactory,
+        ngZone,
+        overlay,
+        positionFactory,
+        scrollDispatcher,
+        viewContainerRef,
+        viewportRuler);
+  }
+
+  protected initFocusTrap(): void {
+    this.focusTrap = this.focusEscapeNotifierFactory.create(this.overlayRef!.overlayElement);
```

Comment:
Should this be cleaned up on destroy?

Reply:
The Overlay elements are destroyed, and the focus trap is inside of the overlay bounding box, so I don't think any additional cleanup is needed.